PayPalMessaging/src/main/java/com/braintreepayments/api/PayPalMessagingListener.kt

```diff
@@ -0,0 +1,34 @@
+package com.braintreepayments.api
+
+/**
+ * Callback used to communicate [PayPalMessagingView] events
+ * Note: **This module is in beta. It's public API may change or be removed in future releases.**
+ */
+interface PayPalMessagingListener {
+
+    /**
+     * Called when the user has clicked the [PayPalMessagingView].
+     */
+    fun onClick()
+
+    /**
+     * Called when the user has begun the PayPal Credit application.
+     */
+    fun onApply()
+
+    /**
+     * Called when the [PayPalMessagingView] is loading.
+     */
+    fun onLoading()
+
+    /**
+     * Called when the [PayPalMessagingView] has rendered successfully.
+     */
+    fun onSuccess()
```

Comment:
Is there a callback for the merchant when the PayPalMessagingView is closed/completed?

Reply:
The messaging SDK does not provide a hook into when the view is closed/completed currently